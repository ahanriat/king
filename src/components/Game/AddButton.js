import ReactNative from 'react-native';
import React from 'react';

const {
  StyleSheet,
  View,
  Text,
  TouchableOpacity,
} = ReactNative;

const {
  PropTypes,
  Component,
} = React;

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
    flexDirection: 'column',
  },
  content: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
    flexDirection: 'column',
  },
  text: {
    fontSize: 30,
    color: 'white',
  },
});

class AddButton extends Component {

  render() {
    return (
      <View style={this.props.style}>
        <View style={styles.container}>
          <TouchableOpacity
            onPress={this.props.onPress}
          >
            <View style={styles.content}>
              <Text style={styles.text}>{this.props.text}</Text>
            </View>
          </TouchableOpacity>
        </View>
      </View>
    );
  }

}

AddButton.propTypes = {
  onPress: PropTypes.func,
  text: PropTypes.string,
  style: PropTypes.number,
};

module.exports = AddButton;
