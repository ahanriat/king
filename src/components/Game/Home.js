import React from 'react';
import ReactNative from 'react-native';

const {
  StyleSheet,
  View,
  Text,
} = ReactNative;

const {
  Component,
} = React;

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  title: {
    color: '#ffffff',
  },

});

class GameHome extends Component {

  render() {
    return (
      <View style={styles.container}>
        <Text style={styles.title}>Ok</Text>
      </View>
    );
  }

}

GameHome.propTypes = {
};

export default GameHome;
