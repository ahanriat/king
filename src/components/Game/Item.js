import React from 'react';
import ReactNative from 'react-native';

const {
  StyleSheet,
  View,
  Text,
  TouchableOpacity,
  Animated,
} = ReactNative;

const {
  PropTypes,
  Component,
} = React;

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'flex-start',
    paddingLeft: 30,
    marginTop: 15,
    marginBottom: 15,
    height: 40,
  },
  titleWrapper: {
    flex: 1,

  },
  title: {
    fontFamily: 'Montserrat-Light',
    fontSize: 19,
    color: '#ffffff',
  },

  subTitleWrapper: {
    flex: 1,
    flexDirection: 'row',
  },
  winner: {
    fontFamily: 'Montserrat-Light',
    color: '#ffffff',
  },
  looser: {
    fontFamily: 'Montserrat-Light',
    color: '#ffffff',
  },
});
const ACTION_TIMER = 400;

class GameList extends Component {

  constructor() {
    super();

    this.state = {
      pressAction: new Animated.Value(0),
    };
  }

  componentWillMount() {
    this.progressValue = 0;
    this.state.pressAction.addListener((v) => { this.progressValue = v.value; });
  }

  handlePressIn() {
    Animated.timing(this.state.pressAction, {
      duration: ACTION_TIMER,
      toValue: 1,
    }).start(() => this.animationActionComplete());
  }

  handlePressOut() {
    if (this.progressValue < 1) {
      this.props.onSelect(this.props.game.getId());
    }
    Animated.timing(this.state.pressAction, {
      duration: 0,
      toValue: 0,
    }).start();
  }

  animationActionComplete() {
    if (this.progressValue === 1) {
      this.props.onLongPress(this.props.game.getId());
    }
  }

  renderPlayers() {
    const { game } = this.props;

    if (game.getWinnerId()) {
      return (
        <View style={styles.subTitleWrapper}>
          <View style={styles.winnerWrappr}>
            <Text style={styles.winner}>{game.getWinnerId()}</Text>
          </View>
          <View style={styles.looserWrapper}>
            <Text style={styles.looser}>{game.getLooserId()}</Text>
          </View>
        </View>
      );
    }
    return false;
  }

  render() {
    const { game } = this.props;
    return (
      <View style={styles.container}>
        <TouchableOpacity
          style={styles.closeButton}
          onPressIn={() => this.handlePressIn()}
          onPressOut={() => this.handlePressOut()}
        >
          <View style={styles.titleWrapper}>
            <Text style={styles.title}>{game.getTitle()}</Text>
          </View>

          {this.renderPlayers()}
        </TouchableOpacity>
      </View>
    );
  }

}

//

GameList.propTypes = {
  game: PropTypes.object,
  onSelect: PropTypes.func,
  onLongPress: PropTypes.func,
};

module.exports = GameList;
