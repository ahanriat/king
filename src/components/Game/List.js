import React from 'react';
import ReactNative from 'react-native';

import GameItem from './Item';

const {
  StyleSheet,
  View,
  ScrollView,
} = ReactNative;

const {
  PropTypes,
  Component,
} = React;

const styles = StyleSheet.create({
  container: {

  },

});

class GameList extends Component {
  render() {
    const { games, onItemSelected, onItemLongPress } = this.props;
    const sortedGames = games.sortBy((value) => {
      return parseInt(value.get('id'), 10);
    }).reverse();
    return (
      <View style={this.props.style}>
        <ScrollView style={styles.container}>
          {sortedGames.valueSeq().map((game) => {
            return (
              <GameItem
                onLongPress={onItemLongPress}
                onSelect={onItemSelected}
                key={game.getId()}
                game={game}
              />
            );
          })}
        </ScrollView>
      </View>

    );
  }

}

GameList.propTypes = {
  style: PropTypes.number,
  games: PropTypes.object,
  onItemSelected: PropTypes.func,
  onItemLongPress: PropTypes.func,
};

module.exports = GameList;
