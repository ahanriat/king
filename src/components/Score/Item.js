import React from 'react';
import ReactNative from 'react-native';

const {
  StyleSheet,
  View,
  Text,
} = ReactNative;

const {
  PropTypes,
  Component,
} = React;

const styles = StyleSheet.create({
  container: {
    marginBottom: 30,
    height: 50,
    justifyContent: 'center',
    alignItems: 'center',
  },
  score: {
    fontFamily: 'Montserrat-Regular',
    color: '#FFFFFF',
    fontSize: 22,
    textAlign: 'center',
    opacity: 0.5,
  },

  scoreWinner: {
    color: '#F8E71C',
    opacity: 1,
  },
});

class GameItem extends Component {

  render() {
    const { score } = this.props;
    const stylesScore = [styles.score];
    if (score.getValue() === 0) {
      stylesScore.push(styles.scoreWinner);
    }
    return (
      <View
        style={styles.container}
      >
        <Text style={stylesScore}>{score.getValue()}</Text>
      </View>
    );
  }

}

//

GameItem.propTypes = {
  score: PropTypes.object,
};

module.exports = GameItem;
