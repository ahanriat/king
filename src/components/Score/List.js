import React from 'react';
import ReactNative from 'react-native';

import ScoreItem from './Item';

const {
  StyleSheet,
  View,
  ScrollView,
  Text,
} = ReactNative;

const {
  PropTypes,
  Component,
} = React;

const styles = StyleSheet.create({
  scrollviewWrapper: {
    flex: 1,
    flexDirection: 'row',
  },

  container: {
    flex: 1,
    flexDirection: 'row',
    alignItems: 'flex-start',
  },

  roundContainer: {
    width: 50,
  },
  roundTitle: {
    color: '#FFFFFF',
    textAlign: 'center',
  },

  scoreContainer: {

  },
});

class ScoreList extends Component {
  render() {
    const { rounds, players } = this.props;
    const sortedRounds = rounds.sort(
      (a, b) => a.id < b.id
    );
    return (
      <View style={styles.scrollviewWrapper}>
        <ScrollView horizontal>
          <View style={styles.container}>
            { sortedRounds.valueSeq().map((round) => {
              return (
                <View style={styles.roundContainer} key={`round_${round.id}`}>
                  <Text style={styles.roundTitle}>
                    {round.id}
                  </Text>
                  <View style={styles.scoreContainer}>
                    { players.map((player, key) => {
                      const score = round.scores.filter((item) => {
                        return item.getPlayerId() === player;
                      })[0];

                      if (score) {
                        return (
                          <ScoreItem
                            score={score}
                            key={`score_${score.getId()}`}
                          />
                        );
                      }
                      return <View key={`score_empty_${key}`} />;
                    })}
                  </View>

                </View>
              );
            })}
          </View>
        </ScrollView>
      </View>
    );
  }

}

ScoreList.propTypes = {
  style: PropTypes.number,
  rounds: PropTypes.object,
  players: PropTypes.object,
};

module.exports = ScoreList;
