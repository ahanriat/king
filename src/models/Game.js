import { Record, List } from 'immutable';

const GameRecord = Record({
  id: null,
  title: null,
  playerIds: new List(),
  winnerId: null,
  looserId: null,
  createdAt: new Date(),
  closedAt: null,
});

const pad = (num, size) => {
  let s = `${num}`;
  while (s.length < size) s = `0${s}`;
  return s;
};

class Game extends GameRecord {
  getId() {
    return this.get('id');
  }

  getTitle() {
    return `Partie #${pad(this.get('id'), 3)}`;
  }

  getPlayerIds() {
    return this.get('playerIds');
  }

  getWinnerId() {
    return this.get('winnerId');
  }

  getLooserId() {
    return this.get('looserId');
  }

  getCreatedAt() {
    return this.get('createdAt');
  }

  getClosedAt() {
    return this.get('closedAt');
  }
}

module.exports = Game;
