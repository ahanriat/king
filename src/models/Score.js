import { Record } from 'immutable';

const ScoreRecord = Record({
  id: null,
  playerId: null,
  value: 0,
  round: 0,
  gameId: null,
  createdAt: null,
});

class Score extends ScoreRecord {
  getId() {
    return this.get('id');
  }

  getPlayerId() {
    return this.get('playerId');
  }

  getGameId() {
    return this.get('gameId');
  }

  getValue() {
    return this.get('value') || 0;
  }

  getRound() {
    return this.get('round') || 0;
  }
}

module.exports = Score;
