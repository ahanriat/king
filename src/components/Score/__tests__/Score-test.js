import React from 'react';
import { Map, fromJS } from 'immutable';
import renderer from 'react-test-renderer'; // eslint-disable-line

import Score from '../../../models/Score';
import Game from '../../../models/Game';

import ScoreList from '../List';
import ScoreItem from '../Item';

describe('Game', () => {
  it('render List', () => {
    const gameData = {
      id: '1',
      title: 'title game',
      playerIds: fromJS(['id1', 'id2']),
      createdAt: new Date(1987, 11, 12),
      closedAt: new Date(1987, 11, 12),
      winnerId: 'id1',
    };
    const scoreData = {
      id: 'scoreId1',
      playerId: 'playerId1',
      gameId: '1',
      value: 12,
      round: 1,
    };
    const game = new Game(gameData);
    const score = new Score(scoreData);
    let rounds = new Map();
    rounds = rounds.set('0', {
      id: 0,
      scores: [score],
    });
    const props = {
      rounds,
      players: game.getPlayerIds(),
    };
    const tree = renderer.create(
      <ScoreList {...props} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('render Item', () => {
    const data = {
      id: 'scoreId1',
      playerId: 'playerId1',
      gameId: '1',
      value: 12,
      round: 1,
    };
    const score = new Score(data);

    const props = {
      gameId: '1',
      score,
    };
    const tree = renderer.create(
      <ScoreItem {...props} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
