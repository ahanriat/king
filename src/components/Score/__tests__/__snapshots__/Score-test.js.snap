exports[`Game render Item 1`] = `
<View
  style={undefined}>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#FFFFFF",
        "flex": 1,
        "fontFamily": "Montserrat-Light",
        "textAlign": "center"
      }
    }>
    12
  </Text>
</View>
`;

exports[`Game render List 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "row"
    }
  }>
  <ScrollView
    horizontal={true}>
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "flex": 1,
          "flexDirection": "row"
        }
      }>
      <View
        style={
          Object {
            "width": 50
          }
        }>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#FFFFFF",
              "textAlign": "center"
            }
          }>
          0
        </Text>
        <View
          style={Object {}}>
          <View />
          <View />
        </View>
      </View>
    </View>
  </ScrollView>
</View>
`;
